controllers/securitypolicy_controller.go

```diff
@@ -446,8 +449,24 @@ func (r *SecurityPolicyReconciler) updateStatusOfLinkedAPIs(ctx context.Context,
 		Namespace: &namespace, Name: policy.Name,
 	}
 
+	oldLinks := map[string]bool{}
+	newLinks := map[string]bool{}
+
+	for _, t := range policy.Status.LinkedAPIs {
+		oldLinks[t.String()] = true
+	}
+
+	for _, t := range policy.Spec.AccessRightsArray {
+		name := types.NamespacedName{Name: t.Name, Namespace: t.Namespace}
+		newLinks[name.String()] = true
+	}
+
 	// Remove links from api definitions
 	for _, t := range policy.Status.LinkedAPIs {
+		if _, ok := newLinks[t.String()]; ok {
+			continue
+		}
+
 		api := &tykv1.ApiDefinition{}
 
 		namespace := ""
```

Comment:
i know this part is not related to the PR but why we are overriding `namepsace` variable's value to an empty string? It is already declared as `policy.Namespace` at line 446.

Reply:
On line 446, namespace is referring to policy namespace. On line 453 it is referring to API namespace. 
Needs a rename to avoid confusion.